shardingsphere-kernel/shardingsphere-data-pipeline/shardingsphere-data-pipeline-core/src/main/java/org/apache/shardingsphere/data/pipeline/core/job/FinishedCheckJob.java

```diff
@@ -101,15 +100,19 @@ public void execute(final ShardingContext shardingContext) {
     
     private boolean checkJobStatus(final String jobId) {
         Map<Integer, JobProgress> jobProgressMap = ruleAlteredJobAPI.getProgress(jobId);
-        final AtomicBoolean flag = new AtomicBoolean(false);
+        boolean flag = false;
         for (JobProgress each : jobProgressMap.values()) {
+            if (each == null) {
+                continue;
+            }
             if (JobStatus.FINISHED.equals(each.getStatus()) || JobStatus.PREPARING_FAILURE.equals(each.getStatus())
-                    || JobStatus.EXECUTE_INVENTORY_TASK_FAILURE.equals(each.getStatus()) || JobStatus.EXECUTE_INCREMENTAL_TASK_FAILURE.equals(each.getStatus())) {
-                flag.set(true);
+                    || JobStatus.EXECUTE_INVENTORY_TASK_FAILURE.equals(each.getStatus())
+                    || JobStatus.EXECUTE_INCREMENTAL_TASK_FAILURE.equals(each.getStatus())) {
+                flag = true;
                 break;
             }
```

Comment:
It's better to depend on less status as previous comment discussed. Not only simplify code, but also if there's new status added, then it's not necessary to change code here.

`jobProgressMap` for each could be used to reduce code lines.